HW1/P6/P6.py

```diff
@@ -0,0 +1,59 @@
+import numpy as np
+from functools import partial
+import findspark
+findspark.init()
+import pyspark
+
+
+def pre_process_data( filename, sc ):
+    wlist = sc.textFile( filename )
+    wlist_words = wlist.flatMap( lambda x: x.split() )
+    wlist_words_filter = wlist_words.filter( lambda x: not( x.isdigit() or x.isupper() or ( x[:-1].isupper() and x[-1:] == "." ) ) )
+    wlist_words_index = wlist_words_filter.zipWithIndex().map( lambda (x,y): (y,x) )
+    wlist_words_index_2 = wlist_words_index
+    size = wlist_words_index_2.count()
+    wlist_words_index_2 = wlist_words_index_2.map( lambda ( x, y ): ( x-1, y ) if x != 0 else ( size, y ) )
+    wlist_words_2 = wlist_words_index.join( wlist_words_index_2 )
+    wlist_words_index_2 = wlist_words_index_2.map( lambda ( x, y ): ( x-1, y ) if x != 0 else ( size, y ) )
+    wlist_words_final = wlist_words_2.join( wlist_words_index_2 ).map( lambda ( x, y ): ( ( y[0][0], y[0][1], y[1] ) , 1 ) ).reduceByKey( lambda a,b: a + b ).map( lambda ( x, y ): ( ( x[0], x[1] ), [ ( x[2], y ) ] ) ).reduceByKey( lambda a,b: a + b)
```

Comment:
I couldn't figure out how to do this in Spark, so I just used regular Python. Well done.